### PR TITLE
Add thread model

### PR DESCRIPTION
**Pull request by ElijahGeorge**
If this PR fixes a bug, you _must_ add test cases representative of the bug.
#### Screenshots (if appropriate):

### CODE REVIEW

**Comment by jsachs** [comment]
@ElijahGeorge going to change this to WIP for the next few days, since I don't think we want this causing any messaging regressions before Indaba MVP.

File: server/controllers/threads.controller.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * @returns {Message}
+ */
+function create(req, res, next) {
+    const participants = req.body.participants;
+    console.log("found participant", participants)
+    let users = [];
+    let userPromises = [];
+    participants.forEach((participant) => {
+      let userPromise = User.findOrCreate({where: {username: participant}})
+      .spread((user, created) => {
```
Comment (puckybreg) [inline server/controllers/threads.controller.js]: @ElijahGeorge so you are using `.spread` here because sequelize's `findOrCreate` returns an array?

**Review by puckybreg — APPROVED** [review]
Looks Good

File: server/controllers/threads.controller.js (hunk 1 above)
Comment (ElijahGeorge) [reply server/controllers/threads.controller.js]: Yes @puckybreg. `created` is a `boolean` that indicates whether the returned user was created or just found in the database. It's actually different from the spread operator `...`. I'm not sure why they decided to name it that 😕

**Comment by ElijahGeorge** [comment]
Got it @jsachs. Right now it shouldn't break anything that's being used by Indaba as I created new routes and controllers but I agree that it should remain a WIP

**Review by jsachs — APPROVED** [review]

**Comment by nickmccally** [comment]
@ElijahGeorge  I was prepared for that question! In my experience an index of some sort will be on an id, and the sql instance will handle it more efficiently. Since we are joining on the tables by id already, it is highly likely we will either have a key or index on this column already. That would mean potentially one less index, and less index maintenance. I'm not opposed to changing it, it is only a quick column name change in the code... that said, I think this will serve best for now.

**Comment by ElijahGeorge** [comment]
Hmm..great point @nickmccally! I think I agree!

**Comment by B3rry** [comment]
@jsachs Hey, is there anything else we need to do to prepare this branch to be merged into develop?

**Comment by jsachs** [comment]
Just final testing and a review pass from @sirctseb

**Review by sirctseb — APPROVED** [review]

File: server/controllers/threads.controller.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    let date = new Date();
+    participants.forEach((participant) => {
+      let userPromise = User.findOrCreate({where: {username: participant}})
+      .spread((user, created) => {
+        users.push(user);
+      });
+      userPromises.push(userPromise);
+    });
+
+    Promise.all(userPromises).then(() => {
```
Comment (sirctseb) [inline server/controllers/threads.controller.js:35]: Promise.all resolves with an array of the results, so here and above can be shortened to `Promise.all(participants.map(participant => User.findOrCreate({where: {username: participant}}))).then(users => {`

**Pull request merged**